scripts/mcp_inspector_executor.py

```
@@ -1,35 +1,49 @@
     mode = sys.argv[1]
-    command = []
+    command: list[str]
 
     if mode == "http":
-        command = ["mcp-inspector", "-v", "validate", "http://localhost:8000/mcp"]
+        # Start the HTTP server first
+        server_proc = subprocess.Popen(
+            ["python3", "src/adaptive_graph_of_thoughts/main.py"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)